nevergrad/common/decorators.py

```diff
@@ -3,28 +3,36 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, Callable, Optional, Dict
+from collections import UserDict
+from typing import Any, Callable, Optional, Dict, TypeVar
 import functools
 
 
-class Registry(dict):  # type: ignore
+X = TypeVar("X")
+
+
+# pylint does not understand Dict, and mypy needs Dict to be generic...
+# the following is (very) hacky but solves both issues for now
+# It can be safely reverted to Registry(dict) though...
+# pylint: disable=too-many-ancestors
+class Registry(Dict[str, X], UserDict):  # type: ignore
     """Registers function or classes as a dict.
     """
 
     def __init__(self) -> None:
         super().__init__()
         self._information: Dict[str, Dict[Any, Any]] = {}
 
-    def register(self, obj: Any, info: Optional[Dict[Any, Any]] = None) -> Any:
+    def register(self, obj: X, info: Optional[Dict[Any, Any]] = None) -> Any:
         """Decorator method for registering functions/classes
         The info variable can be filled up using the register_with_info
         decorator instead of this one.
         """
-        name = obj.__name__ if hasattr(obj, "__name__") else obj.__class__.__name__
+        name = obj.__name__ if hasattr(obj, "__name__") else obj.__class__.__name__  # type: ignore
```

Comment:
This is `getattr(obj, "__name__", obj.__class__.__name__)`.

Reply:
Oh did not know there was a default!